### ISSUE TITLE
Many-to-many not working

### ISSUE THREAD

**Issue by sirakoff** [balderdashy/sails-mongo#210]
`api/models/Post.js`

```javascript
module.exports = {
  adapter: 'mongo',
  tableName: 'posts',
  autoCreatedAt: true,
  autoUpdatedAt: true,
  attributes: {
	ID: {
		type: 'string',
		unique: true,
		index: true,
	},
	title: {
		type: 'string'
	},
	user:{
		model:'user'
	},
	tags:{
		collection: 'tag',
		via: 'posts'
	},
	createdAt: {
		type: 'datetime',
		defaultsTo: function (){ return new Date(); },
		columnName: 'created_at'
	},
	updatedAt: {
		type: 'datetime',
		defaultsTo: function (){ return new Date(); },
		columnName: 'updated_at'
	},
  }
};
```

`api/models/Tag.js`

```javascript
module.exports = {
  adapter: 'mongo',

  tableName: 'tags',
  attributes: {
	name:{
		type: 'string',
		unique: true,
		index: true
	},
	slug:{
		type: 'string',
		index: true
	},
	posts: {
		collection: 'post',
		via: 'tags'
	}
  }
};
```

**Comment by akshitjain** [comment]
Yes an error is encountered with many to many associations.
When we do this association the join table is created but on reading the error I infer that the attributes aren't created in that table.Error explained below:-
While adding the data to the join table using this query: 
User.findOne(2).exec(function(err, user) {
  if(err) // handle error

  // Queue up a record to be inserted into the join table
  user.pets.add(20);

  // Save the user, creating the new associations in the join table
  user.save(function(err) {});
});

it gives an error mentioning  pets is not created.Please look into this

**Comment by sailsbot** [comment]
Thanks for posting, @sirakoff.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
